doc/newfeat.texi

```diff
@@ -4,117 +4,75 @@
+@code{ess-eval-line-and-step-invisibly} in @code{sas-mode-local-map};
+@code{C-c f}, which was bound to @code{ess-insert-function-outline} in
+@code{ess-add-MM-keys}; and @code{C-c h}, which was bound to
+@code{ess-handy-commands} in @code{Rd-mode-map},
+@code{ess-noweb-minor-mode-map}, and @code{ess-help-mode-map}
+
+@item @ESS{[R]}: @code{ess-r-package-use-dir} now works with any mode.
+This sets the working directory to the root of the current package
+including for example C or C++ files within @code{/src}).
+
 @item @ESS{[R]}: Long + + prompts in the inferior no longer offset output.
 New option @code{strip} for @code{inferior-ess-replace-long+} strips the
 entire + + sequence.
 
-@item @ESS{[R]}: Fontification of roxygen @code{@@param} keywords now
-supports comma-separated parameters.
-
-@item @ESS{[R]}: Function-like keywords such as @code{if ()} or
-@code{stop()} are no longer fontified as keyword if not followed by an
-opening parenthesis. The same holds for search path modifiers like
-@code{library()} or @code{require()}. This feature is only available in
-Emacs >= 25.
-
-@item @ESS{[R]}: We have improved fontification of keywords so they
-better reflect the semantics of the R language. @code{ess-R-keywords}
-now only contains words reserved by the R parser.
-@code{ess-R-control-flow-keywords} contains words of base functions that
-cause non-contiguous control flow, such as @code{return()} and
-@code{stop()}. It includes the following variables that were previously
-not fontified: @code{on.exit()}, @code{tryCatch()},
-@code{withRestarts()}, @code{invokeRestart()}, @code{recover()} and
-@code{browser()}
-
-Finally, @code{ess-R-signal-keywords} contains functions part of the
-condition system that only potentially impact control flow:
-@code{message()}, @code{warning()} (moved from @code{ess-R-keywords}),
-@code{signalCondition()} and @code{withCallingHandlers()}. These
-keywords inherit from @code{ess-modifiers-face} (the face used for
-@code{library()} etc).
-
 @item ESS modes now inherit from @code{prog-mode}.
+In the next release, ESS modes will use @code{define-derived-mode} so
+that each mode will have (for example) its own hooks and keymaps.
 
-@item @ESS{[R]}: The package development minor mode now only activates
-within editing buffers by default, i.e. ones that inherit from
-@code{prog-mode} or @code{text-mode}. If you want to restore the old
-behavior and activate the package mode in all buffers (that have a
-@code{default-directory} that is part of a package path), set
-@code{ess-r-package-auto-activate} to @code{t}.
-
-@item ESS now provides support for flymake in R buffers for Emacs 26 and newer.
-Users need to install the @code{lintr} package (available on CRAN) to
-use it. Customizable options include @code{ess-use-flymake},
-@code{ess-r-flymake-linters}, and @code{ess-r-flymake-lintr-cache}.
+@item @ESS{[R]}: The package development minor mode only activates in file-visiting buffers.
```

Comment:
This one was reverted and is no longer true.

Reply:
OK, removed.